src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -794,6 +794,9 @@ public void run() {
                     if (exec == null) {
                         throw new IllegalStateException("running task without associated executor thread");
                     }
+                    if (!context.hasBody()) {
+                        throw new Exception("node must be called with a body");
```

Comment:
Would it be better to make this the first line of `ExecutorStepExecution.start` so we don't even get to the point of adding a `PlaceholderTask` to the queue? Other than that, I might change the exception type since we don't care about a stack trace, and tweak the message:

```suggestion
                        throw new AbortException("The node step must be called with a body"); // Requires an import of hudson.model.AbortException
```